scipy/optimize/tests/test_linprog.py

```diff
@@ -2366,36 +2366,18 @@ def test_mip_rel_gap_passdown(self):
         assert not np.all(gap_diffs == 0)
 
     def test_semi_continuous(self):
-        # See issue #18106. This test if the integrality parameter
-        # is being checked correctly (status returns 0),
-        # when its > 1 it allows values to be 0 even if its out of bounds.
-
-        c = np.array([-1.])
-        A_ub = np.array([[1.]])
-        b_ub = np.array([0.5])
-        A_eq = np.zeros((0, 1), dtype=float)
-        b_eq = np.array([], dtype=float)
-        bounds = np.array([[1., 5.]])
-        integrality = np.array([2])
+        # See issue #18106. This tests whether the solution is being
+        # checked correctly (status is 0) when integrality > 1:
+        # values are allowed to be 0 even if 0 is out of bounds.
 
-        res = linprog(c, A_ub, b_ub, A_eq, b_eq, bounds,
-                      integrality=integrality, method='highs')
-
-        assert np.all(res.x == 0)
-        assert res.status == 0
+        c = np.array([1., 1., -1, -1])
+        bounds = np.array([[0.5, 1.5], [0.5, 1.5], [0.5, 1.5], [0.5, 1.5]])
+        integrality = np.array([2, 3, 2, 3])
 
-        c = np.array([-1., -1., 1])
-        A_ub = np.array([[1., 1., 0], [1.,1., 0], [0,0,1]])
-        b_ub = np.array([0.5, 0.5, 1.])
-        A_eq = np.zeros((0, 3), dtype=float)
-        b_eq = np.array([], dtype=float)
```

Comment:
`A_eq` and `b_eq` were unused, so no need to create empty arrays.
`A_ub` and `b_ub` added complexity to the problem so that it was just a bit harder to predict the outcome. If they are not strengthening the test somehow, let's simplify things by removing them.